src/interfaces/store-interfaces/state-interfaces/name-request-interface.ts

```diff
@@ -1,43 +1,57 @@
-import { CorpTypeCd, NameRequestStates } from '@/enums'
+import { CorpTypeCd, NameRequestStates, NameRequestTypes } from '@/enums'
 
-/** Name request response details interface. */
-export interface NameRequestDetailsIF {
-  approvedName: string
-  status: NameRequestStates
-  consentFlag: string
-  expirationDate: string
-}
-
-/** Name request applicant details interface. */
+/**
+ * Name request applicant interface.
+ * Includes only the properties we care about.
+ */
 export interface NameRequestApplicantIF {
+  addrLine1: string
+  addrLine2: string
+  addrLine3: string
+  city: string
+  countryTypeCd: string
+  emailAddress: string
   firstName: string
-  middleName: string
   lastName: string
-  emailAddress: string
+  middleName: string
   phoneNumber: string
-  addressLine1: string
-  addressLine2: string
-  addressLine3: string
-  city: string
-  countryTypeCode: string
-  postalCode: string
-  stateProvinceCode: string
+  postalCd: string
+  stateProvinceCd: string
 }
 
-/** Name Request state interface. */
+/**
+ * Name request name interaface.
+ * Includes only the properties we care about.
+ */
+export interface NameRequestNameIF {
+  name: string
+  state: NameRequestStates
+}
+
+/**
+ * Name request interface.
+ * Includes only the properties we care about.
+ */
 export interface NameRequestIF {
-  nrNumber: string
-  entityType: CorpTypeCd
-  details: NameRequestDetailsIF
-  applicant: NameRequestApplicantIF
-  filingId: number
+  applicants: NameRequestApplicantIF
+  consentFlag: string
+  expirationDate: string // yyyy-mm-ddRhh:mm:ss+00:00
+  legalName?: string // from IA only
+  legalType: CorpTypeCd
+  names: Array<NameRequestNameIF>
+  nrNum: string
```

Comment:
Note: in the NR object, the name request number is `nrNum`.

However, when a pseudo NR object is in an IA or reg filing, then the name request numbers is `nrNumber`.